veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,178 @@
+One such example is SR-IOV free VFs that must be available on a node for the VM to make use of.
+The names of those custom resources are listed as an annotation in the network-attachment-definition (NAD) CR that corresponds to the secondary network requested by the VM.
+
+Currently, virt-controller queries the NAD CR for every secondary network in every VM that is launched.
+Network-attachment-definition is an external project maintained by the k8s network plumbing WG. The mentioned query is the only direct dependency that KubeVirt production code has on this external project.
+This proposal aims to remove that query and code dependency in favor of an external replacement that will map the custom resources to the virt-launcher pod instead.  
+In addition, the virt-controller role binding that currently allows it to access the NAD can be reduced in scope.
+
+
+## Motivation
+
+The replacement has the architectural advantage of decoupling KubeVirt from an external dependency by simplifying code maintenance and supporting future replacement technologies.
+Security will be improved by removing access permissions to the external CRD.
+In addition, a performance improvement can be gained from using a project that caches NAD data. 
+    
+
+## Goals
+
+NAD query code will be deleted from virt-controller and a replacement injector will be introduced. 
+
+## Non Goals
+
+Removal of other multus related dependencies 
+
+## Definition of Users
+
+This enhancement is intended for:
+- VM owners
+- Namespace owners
+
+## User Stories
+
+- As a VM owner, I want to accelerate VM load time
+- As a VM owner and a namespace owner, I want my system to use the principle of least privilege, such that service accounts will only have access to resources that they absolutely need. 
+
+## Repos
+
+https://github.com/kubevirt/kubevirt
+https://github.com/k8snetworkplumbingwg/network-resources-injector
+
+## Design
+
+### Current Architecture
+KubeVirt currently has a tight coupling with NetworkAttachmentDefinition (NAD) resources. When a VM with secondary networks is launched, the virt-controller directly queries NAD CRs to extract custom resource requirements ( `k8s.v1.cni.cncf.io/resourceName` annotation) and injects them into the virt-launcher pod specification, as resource requests and limit.
+
+### Proposed Architecture
+The new architecture decouples KubeVirt from NAD resources by leveraging the network-resources-injector as an external admission controller. This creates a cleaner separation of concerns where KubeVirt focuses on VM lifecycle management while network-resources-injector handles network resource injection.
+
+### Deployment alternatives 
+
+#### Option 1: Deprecate existing injection code and require manual deployment of network-resources-injector
+
+Current KubeVirt functionality will be deprecated and removed within 3 KubeVirt releases.
```

Comment:
How will KubeVirt decide whether to perform the query and add the resources to the virt-launcher pod?